jupyterhub_moss/spawner.py

```diff
@@ -290,35 +292,24 @@ async def create_options_form(spawner):
     _MEM_REGEXP = re.compile("^[0-9]*([0-9]+[KMGT])?$")
 
     def __validate_options(self, options):
-        """Check validity of options"""
+        """Check validity/syntax of options"""
         assert "partition" in options, "Partition information is missing"
         assert options["partition"] in self.partitions, "Partition is not supported"
 
-        partition_info = self.partitions[options["partition"]]
-
         if "runtime" in options:
-            runtime = parse_timelimit(options["runtime"])
-            assert (
-                runtime.total_seconds() <= partition_info["max_runtime"]
-            ), "Requested runtime exceeds partition time limit"
+            parse_timelimit(options["runtime"])  # Raises exception if malformed
 
-        if (
-            "nprocs" in options
-            and not 1 <= options["nprocs"] <= partition_info["max_nprocs"]
-        ):
-            raise AssertionError("Error in number of CPUs")
+        if "nprocs" in options and not 1 <= options["nprocs"]:
```

Comment:
Advantage of keeping as much checks as possible here is that exception messages are displayed to the user as opposed to exception raised in `start`.